services/src/main/java/edu/unc/lib/dl/cdr/services/processing/SetAsPrimaryObjectService.java

```diff
@@ -63,6 +68,10 @@ public void setAsPrimaryObject(AgentPrincipals agent, PID fileObjPid) {
         }
         WorkObject work = (WorkObject) parent;
         work.setPrimaryObject(fileObjPid);
+
+        // Send message that the action completed
+        operationsMessageSender.sendIndexingOperation(agent.getUsername(), Arrays.asList(fileObjPid),
```

Comment:
Its been a little while, but weren't we thinking of making an event specific to this action rather than sending an indexing operation?

Reply:
hmm, i don't remember what we decided. i recall you saying at one point that it was currently handled this way and that it would be a fine starting point. i can create another operation, though.